internal/controller/proposedcommit_controller.go

```diff
@@ -98,10 +98,55 @@ func (r *ProposedCommitReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, err
 	}
 
-	branchShas, err := gitOperations.GetBranchShas(ctx, []string{pc.Spec.ActiveBranch, pc.Spec.ProposedBranch})
+	err = r.calculateStatus(ctx, &pc, gitOperations)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	err = r.creatOrUpdatePullRequest(ctx, &pc)
+	if err != nil {
+		return ctrl.Result{}, err
```

Comment:
No requeue specified means requeue immediately, right? Probably need to either build a retry/backoff into the function, or set a nonzero requeue here. Doesn't have to be in this PR, a TODO would be fine.

Reply:
By default it's an exponential backoff